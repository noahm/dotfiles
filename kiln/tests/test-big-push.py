#!/usr/bin/env python
#
# Test big-push extension

import os
import common

import hgtest
import kilntest

hgt = common.BfilesTester()

hgt.updaterc({'extensions': [('kilnauth', kilntest.KILNAUTHPATH),
                             ('big-push', kilntest.BIGPUSHPATH)]})
hgt.announce('setup')
token = kilntest.gettoken()
kilntest.deletetest(hgt, token)
test = kilntest.createtest(hgt, token)
hgt.hg(['clone', kilntest.KILNURL + '/Repo/Test/Test/Test', 'repo1'], log=False,
        stdout='''no changes found
updating to branch default
0 files updated, 0 files merged, 0 files removed, 0 files unresolved
''')
os.chdir('repo1')
hgt.writefile('n1', 'n1')
hgt.hg(['add', 'n1'])
hgt.hg(['commit', '-m', 'add file'])
hgt.hg(['push'], stdout='''pushing to %s/Repo/Test/Test/Test
searching for changes
searching for changes
remote: kiln: successfully pushed one changeset
''' % kilntest.KILNURL)
hgt.writefile('n2', 'n2')
hgt.hg(['add', 'n2'])
hgt.hg(['commit', '-m', 'add another file'])
hgt.hg(['push', '--chunked'], stdout='''pushing to %s/Repo/Test/Test/Test
searching for changes
searching for changes
searching for changes
searching for changes
remote: kiln: successfully pushed one changeset
''' % kilntest.KILNURL)
for i in range(3, 25):
    hgt.writefile('n%d' % i, 'dummy')
    hgt.hg(['add', 'n%d' % i])
    hgt.hg(['commit', '-m', 'changeset %d' % i])
hgt.hg(['push', '--chunked'], stdout='''pushing to %s/Repo/Test/Test/Test
searching for changes
searching for changes
searching for changes
searching for changes
remote: kiln: successfully pushed one changeset
searching for changes
searching for changes
remote: kiln: successfully pushed 2 changesets
searching for changes
searching for changes
remote: kiln: successfully pushed 4 changesets
searching for changes
searching for changes
remote: kiln: successfully pushed 8 changesets
searching for changes
searching for changes
remote: kiln: successfully pushed 7 changesets
''' % kilntest.KILNURL)
hgt.hg(['pull'], stdout='''pulling from %s/Repo/Test/Test/Test
searching for changes
no changes found
''' % kilntest.KILNURL)
hgt.hg(['push'], stdout='''pushing to %s/Repo/Test/Test/Test
searching for changes
searching for changes
no changes found
''' % kilntest.KILNURL)
os.chdir('..')
hgt.hg(['clone', kilntest.KILNURL + '/Repo/Test/Test/Test', 'repo2'], log=False,
       stdout='''requesting all changes
adding changesets
adding manifests
adding file changes
added 24 changesets with 24 changes to 24 files
updating to branch default
24 files updated, 0 files merged, 0 files removed, 0 files unresolved
''')
