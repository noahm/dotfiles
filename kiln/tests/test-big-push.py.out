% setup
hg add n1
hg commit -m 'add file'
hg push
hg add n2
hg commit -m 'add another file'
hg push --chunked
hg add n3
hg commit -m 'changeset 3'
hg add n4
hg commit -m 'changeset 4'
hg add n5
hg commit -m 'changeset 5'
hg add n6
hg commit -m 'changeset 6'
hg add n7
hg commit -m 'changeset 7'
hg add n8
hg commit -m 'changeset 8'
hg add n9
hg commit -m 'changeset 9'
hg add n10
hg commit -m 'changeset 10'
hg add n11
hg commit -m 'changeset 11'
hg add n12
hg commit -m 'changeset 12'
hg add n13
hg commit -m 'changeset 13'
hg add n14
hg commit -m 'changeset 14'
hg add n15
hg commit -m 'changeset 15'
hg add n16
hg commit -m 'changeset 16'
hg add n17
hg commit -m 'changeset 17'
hg add n18
hg commit -m 'changeset 18'
hg add n19
hg commit -m 'changeset 19'
hg add n20
hg commit -m 'changeset 20'
hg add n21
hg commit -m 'changeset 21'
hg add n22
hg commit -m 'changeset 22'
hg add n23
hg commit -m 'changeset 23'
hg add n24
hg commit -m 'changeset 24'
hg push --chunked
hg pull
hg push
